src/Features/CSharp/Portable/RemoveUnusedVariable/CSharpRemoveUnusedVariableCodeFixProvider.cs

```diff
@@ -69,5 +69,19 @@ protected override void RemoveOrReplaceNode(SyntaxEditor editor, SyntaxNode node
 
         protected override SeparatedSyntaxList<SyntaxNode> GetVariables(LocalDeclarationStatementSyntax localDeclarationStatement)
             => localDeclarationStatement.Declaration.Variables;
+
+        protected override bool ShouldOfferFix(ISyntaxFactsService syntaxFacts, SyntaxNode node)
+        {
+            // Local declarations must be parented by an executable block (or global statement) otherwise
```

Comment:
interesting.  i'm surprised we can remove it from a GlobalStatement (as i assume taht would leave the global statement without an actual statement, which i would think woudl break too).  Or is that because when we remove the node we actually walk up and remove the global statement?

Reply:
Like 65 of this file handles that, but removing the whole global statement instead of just the statement. Can't do that here, because removing the if statement would be odd.